sqrl-planner/sqrl-planner-local/src/main/java/com/datasqrl/graphql/inference/SchemaInference.java

```diff
@@ -130,11 +134,25 @@ private InferredField resolveQueryFromSchema(FieldDefinition fieldDefinition,
     Preconditions.checkState(sqrlTable.isPresent(),
         "Could not find associated SQRL type for field %s on type %s",
         fieldDefinition.getName(), fieldDefinition.getType());
-    SQRLTable table = sqrlTable.get();
+    SQRLTable table = resolveSQRLTable(fieldDefinition.getType(), fieldDefinition.getName(), "Query");
 
     return inferObjectField(fieldDefinition, table, fields, parent);
   }
 
+  private SQRLTable resolveSQRLTable(Type fieldType, String fieldName, String rootType) {
```

Comment:
This isn't a generic resolution for all SQRL tables, only the root Query and Subscription fields. Anything nested gets resolved by field name rather than type name.

Reply:
renamed to "root"